### PR TITLE
pass compile environment into build script

### PR DESCRIPTION
**Pull request by dae**
On Windows, MSVC requires INCLUDE to be set to the path of the standard
include files. Not sure if this is the best solution, but it solved issues with
missing includes mentioned in #423 for me.

### CODE REVIEW

File: cargo/cargo_build_script.bzl [hunk 1]
```diff
@@ -52,6 +52,8 @@ def _cargo_build_script_run(ctx, script):
     # Pull in env vars which may be required for the cc_toolchain to work (e.g. on OSX, the SDK version).
     # We hope that the linker env is sufficient for the whole cc_toolchain.
     cc_toolchain, feature_configuration = get_cc_toolchain(ctx)
+    cc_env = get_cc_compile_env(cc_toolchain, feature_configuration)
+    env.update(**cc_env)
```
Comment (damienmg) [inline cargo/cargo_build_script.bzl:56]: I am not sure either this is the best solution but I think we are better with it than without.

```suggestion
    # TODO: Trim down the list of environment shipped.
    env.update(**cc_env)
```

**Review by damienmg — APPROVED** [review]
I just have a recommendation to add a TODO.

Thanks a lot for all your changes, hopefully I can get to them all now.

File: cargo/cargo_build_script.bzl (hunk 1 above)
Comment (dae) [reply cargo/cargo_build_script.bzl:56]: If you'd prefer to keep env changes to a minimum, it seems as if INCLUDE is sufficient for the CC-using crates in my project to compile:

```diff
diff --git a/cargo/cargo_build_script.bzl b/cargo/cargo_build_script.bzl
index 0725c4c..a13826b 100644
--- a/cargo/cargo_build_script.bzl
+++ b/cargo/cargo_build_script.bzl
@@ -53,7 +53,9 @@ def _cargo_build_script_run(ctx, script):
     # We hope that the linker env is sufficient for the whole cc_toolchain.
     cc_toolchain, feature_configuration = get_cc_toolchain(ctx)
     cc_env = get_cc_compile_env(cc_toolchain, feature_configuration)
-    env.update(**cc_env)
+    includes = cc_env.get("INCLUDE")
+    if includes:
+        env["INCLUDE"] = includes
     _, _, linker_env = get_linker_and_args(ctx, cc_toolchain, feature_configuration, None)
     env.update(**linker_env)
```

Would you like me to push this change?

Comment (dae) [reply cargo/cargo_build_script.bzl]: I've tentatively made this change and rebased on master.

**Review by damienmg — APPROVED** [review]

**Comment by damienmg** [comment]
Thanks that's perfect like this.

**Pull request merged**